src/compiler/checker.ts

```diff
@@ -39692,18 +39693,18 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     function checkImportType(node: ImportTypeNode) {
         checkSourceElement(node.argument);
 
-        if (node.assertions) {
-            const override = getResolutionModeOverrideForClause(node.assertions.assertClause, grammarErrorOnNode);
-            if (override) {
-                if (!isNightly()) {
-                    grammarErrorOnNode(node.assertions.assertClause, Diagnostics.resolution_mode_assertions_are_unstable_Use_nightly_TypeScript_to_silence_this_error_Try_updating_with_npm_install_D_typescript_next);
-                }
-                if (getEmitModuleResolutionKind(compilerOptions) !== ModuleResolutionKind.Node16 && getEmitModuleResolutionKind(compilerOptions) !== ModuleResolutionKind.NodeNext) {
-                    grammarErrorOnNode(node.assertions.assertClause, Diagnostics.resolution_mode_assertions_are_only_supported_when_moduleResolution_is_node16_or_nodenext);
-                }
+        if (node.attributes) {
+            const override = getResolutionModeOverride(node.attributes?.attributes, grammarErrorOnNode);
+            const errorNode = node.attributes?.attributes;

```

Comment:
It is pretty weird that you have to write `attributes?.attributes` constantly to actually get at the array, but I don't have a better suggestion.

Reply:
`typeAttributes`, `attributesClause`?